core/src/test/java/org/mule/tck/SensingNullMessageProcessor.java

```diff
@@ -6,34 +6,53 @@
  */
 package org.mule.tck;
 
+import org.mule.NonBlockingVoidMuleEvent;
 import org.mule.VoidMuleEvent;
+import org.mule.api.MessagingException;
 import org.mule.api.MuleEvent;
 import org.mule.api.MuleException;
 import org.mule.api.processor.MessageProcessor;
 import org.mule.api.source.MessageSource;
+import org.mule.processor.AbstractNonBlockingMessageProcessor;
 import org.mule.util.ObjectUtils;
 import org.mule.util.concurrent.Latch;
 
-public class SensingNullMessageProcessor implements MessageProcessor
+import java.util.concurrent.Executors;
+
+public class SensingNullMessageProcessor extends AbstractNonBlockingMessageProcessor implements MessageProcessor
 {
     public MuleEvent event;
     protected InternalMessageSource source = new InternalMessageSource();
     private long waitTime = 0;
     public Latch latch = new Latch();
 
-    public MuleEvent process(MuleEvent event) throws MuleException
+    @Override
+    protected void processNonBlocking(final MuleEvent event, NonBlockingCompletionHandler completionHandler) throws MuleException
     {
-        if (waitTime > 0)
+        sleepIfNeeded();
```

Comment:
This is used in AsyncRequestReplyRequesterTestCase I can't remember exactly why but reworking AsyncRequestReplyRequesterTestCase to not use a sleep is out of scope of this PR.